src/Microsoft.AspNetCore.JsonPatch/Helpers/ExpressionHelpers.cs

```diff
@@ -51,10 +54,35 @@ private static string GetPath(Expression expr, bool firstTime)
                     if (ContinueWithSubPath(memberExpression.Expression.NodeType, false))
                     {
                         var left = GetPath(memberExpression.Expression, false);
+
+                        // if there's a JsonProperty attribute, we must return the PropertyName
+                        // from the attribute rather than the member name 
+                        var jsonPropertyAttribute =
+                            memberExpression.Member.GetCustomAttributes(
+                            typeof(JsonPropertyAttribute), false);
+
+                        if (jsonPropertyAttribute.Count() > 0)
+                        {
+                            // get value
+                            var castedAttribrute = jsonPropertyAttribute.First() as JsonPropertyAttribute;
```

Comment:
Can't use indexing on IEnum<T> apparently, but I changed the cast to a c-type cast.